spec/loophp/collection/Iterator/ResourceIteratorSpec.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\loophp\collection\Iterator;
+
+use InvalidArgumentException;
+use loophp\collection\Iterator\ResourceIterator;
+use PhpSpec\ObjectBehavior;
+
+class ResourceIteratorSpec extends ObjectBehavior
+{
+    public function it_can_iterate(): void
+    {
+        $this->beConstructedWith(fopen('data://text/plain,ABCD', 'rb'));
+
+        $this->getInnerIterator()->shouldIterateAs(['A', 'B', 'C', 'D']);
+    }
+
+    public function it_does_not_allow_non_resource(): void
+    {
+        $this->beConstructedWith(false);
+
+        $this->shouldThrow(InvalidArgumentException::class)->duringInstantiation();
+    }
+
+    public function it_does_not_allow_unsupported_resource(): void
+    {
+        $this->beConstructedWith(imagecreate(100, 100));
+
+        $this->shouldThrow(InvalidArgumentException::class)->duringInstantiation();
+    }
+
+    public function it_is_initializable(): void
+    {
+        $file = fopen(__DIR__ . '/../../../fixtures/sample.txt', 'rb');
+
+        $this->beConstructedWith($file);
+
+        $this->shouldHaveType(ResourceIterator::class);
+    }
```

Comment:
I'm not too familiar with PHPSpec, though I use PHPUnit + Prophecy so some things look familiar 😄 . Let me know if this looks good or I could be doing something better based on best practices.

I wanted to ensure that the change I made [here](https://github.com/loophp/collection/pull/90/files#diff-be52b0e1d1956fe3eefb225a24ccb14b0ffd5c769563b74084f74f892d80678fR33) is tested

Reply:
Seems to be nice at first sight!